pennylane/beta/plugins/default_tensor.py

```diff
@@ -568,77 +633,92 @@ def _get_operator_matrix(self, operation, par):
             return self._array(A, dtype=self.C_DTYPE)
         return self._asarray(A(*par), dtype=self.C_DTYPE)
 
-    def ev(self, obs_nodes, wires):
+    def ev(self, obs_nodes, wires, contraction_method="auto"):
```

Comment:
This function doesn't actually use `contraction_method`. Note that `contraction_method` is for when you are contracting a multi-node network (and there might be different algorithms for which order you contract in). Here, we are just contracting between two nodes at a time (the state and the observable), so no method is specified.

I suppose we could build a network out of `ket`, the `obs_nodes`, and `bra`, then try to find some nice contraction ordering for that. However, I'm uncertain if there is any real advantage to be gained there :thinking:.

Reply:
Another option is to create an outer product of all observables in a single dense node, then use `contract_between` on that node and `ket` (instead of the iteration over local observables that is currently done). Then we would contract that newly contracted tensor with `bra`. But I have a suspicion it's better to do the local contractions instead

Reply:
Thanks, I guess this is just me needing to get up to speed on the inner workings :thinking: